Add Imagebitamp Extensions

kakukogou opened the pull request:
Merge the [ImageBitmap-Extension](https://github.com/kakukogou/spec-imagebitmap-extension) into this specification with some modifications of wording and formatting.

@ChiahungTai, @smaug, @rocallahan and @anssiko, please have a look, thanks.

Commit:
ImageBitmap-extensions.

Commit:
Get rid of Gecko-specified modifiers.

kakukogou commented:
Address issue #4.

anssiko commented:
LGTM with an editorial suggestion:

Rename "`ImageBitmap` extensions" and "`ImageBitmapFactories` extensions" sections to "`ImageBitmap` interface" and "`ImageBitmapFactories` interface" respectively for consistency with the rest of the spec.

These sections are nested within the "ImageBitmap extensions" top-level section, so they inherit the "extensions" part of it.

Thanks for folding the two specs together into the W3C space.

Commit:
Rename the section names to avoid confusion.
ImageBitmap extensions -> ImageBitmap interface.
mageBitmapFactories extensions -> ImageBitmapFactories interface.

kakukogou commented:
Since the "ImageBitmap Extensions" part has been reviewed at the [original repository](https://github.com/kakukogou/spec-imagebitmap-extension) and on some little modification are applied for merging, I would like to merge this PR for now.
